### PR TITLE
Add LiquidFreeSettings Check

### PR DESCRIPTION
**Pull request by navdeep5**
## What are you adding in this PR?

Resolves https://github.com/Shopify/theme-tools/issues/558

This PR introduces enhanced validation checks for `Settings`; specifically ensuring that they do not contain liquid logic in the format of `{% ... %}`.
image

## What's next? Any followup issues?

With this being a new check, we will want to update https://github.com/Shopify/shopify-dev so users can understand their errors. Developer Docs changes: https://github.com/Shopify/shopify-dev/pull/50841

## What did you learn?

When writing this check I thought of three different implementations that could have been used -- it is important to understand the balance of complexity and shipping fast. 

## Before you deploy

- [x] This PR includes a new checks or changes the configuration of a check
  - [x] I included a minor bump `changeset`
  - [x] It's in the `allChecks` array in `src/checks/index.ts`
  - [x] I ran `yarn build` and committed the updated configuration files
    
    
    
    - [ ] If applicable, I've updated the `theme-app-extension.yml` config

- [x] I included a minor bump `changeset`
- [x] My feature is backward compatible

- [ ] I included a patch bump `changeset`

### CODE REVIEW

**Review by miazbikowski — CHANGES_REQUESTED** [review]
🎩 
1. preset root level settings ✅ 
2. preset blocks' (first level) (as hash) settings ✅ 
3. nested preset blocks (as hash) settings ✅ 
4. preset blocks' (first level) (as array) settings ✅ 
5. nested preset blocks (as array) settings ✅

File: packages/theme-check-common/src/checks/liquid-free-settings/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            }
+          ]
+        }
+        {% endschema %}
+      `,
+      };
+
+      const offenses = await check(theme, [LiquidFreeSettings]);
+      expect(offenses).to.have.length(0);
+    });
```
[the remark this answers is not in the document]
Comment (navdeep5) [reply packages/theme-check-common/src/checks/liquid-free-settings/index.spec.ts:29]: This is what I heard too -- still seems to be up in the air. That being said, if you feel as though it makes sense to ignore it in this check, we can proceed with that.

File: packages/theme-check-common/src/checks/liquid-free-settings/index.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          node.blockEndPosition.start,
+        );
+
+        const jsonFile = toJSONAST(jsonString);
+        if (jsonFile instanceof Error) return;
+
+        visit<SourceCodeType.JSON, void>(jsonFile, {
+          Property(schemaNode, ancestors) {
+            if (isInArrayWithParentKey(ancestors, 'settings') && isLiteralNode(schemaNode.value)) {
+              const value = schemaNode.value.value;
```
Comment (albchu) [inline packages/theme-check-common/src/checks/liquid-free-settings/index.ts:39]: Nit: You could destructure both `value` and `loc` out of `schemaNode.value` so your message indicies could read a bit more concise down the line.
```suggestion
              const {value, loc} = schemaNode.value.value;
```

**Review by albchu — APPROVED** [review]

File: packages/theme-check-common/src/checks/liquid-free-settings/index.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (navdeep5) [reply packages/theme-check-common/src/checks/liquid-free-settings/index.spec.ts:29]: After chatting with @stephanie-shopify, `available_if` is up in the air and could be altered at any time, so for safe being, we will ignore liquid in `available_if`.

**Review by charlespwd — APPROVED** [review]

**Pull request merged**